File: Backend/src/main/java/org/example/backend/domain/PreferredTime.java
package org.example.backend.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

@Getter
@Setter
@Entity
@Table(name = "preferred_time")
public class PreferredTime {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ColumnDefault("nextval('preferred_time_id_seq')")
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "day", nullable = false, length = 10)
    private String day;

    @Column(name = "hour", nullable = false)
    private Short hour;

}

File: TTSolver/Model/PreferredTime.py
class PreferredTime:
    def __init__(self, id: int, day: str, hour: int):
        self.id = id
        self.day = day
        self.hour = hour

    def __str__(self):
        return f"PreferredTime(id={self.id}, day={self.day}, hour={self.hour})"
